### PR TITLE
Implement phase one of the migration to group attributes

### PR DESCRIPTION
**Pull request by Perksey**
Implements the new schema from #335 and deprecates the old one. That issue should be kept open until the Khronos Group decides to remove the old group blocks permenantly. 

The enumerants have been tagged using an automated process. As such, the groups will have been preserved in this new format exactly as they were in the old format.

The RNC schema has also been updated.

### CODE REVIEW

**Comment by Perksey** [comment]
This is ready for review.

File: xml/registry.rnc [hunk 1]
```diff
@@ -62,26 +61,9 @@ Type = element type {
     text
 }
 
-# <groups> defines a group of enum groups
-Groups = element groups {
```
Comment (oddhack) [inline xml/registry.rnc]: I wouldn't think the updated XML would validate against the schema with this stuff removed, while the group tags themselves are marked deprecated but still present?

Comment (Perksey) [reply xml/registry.rnc]: Fixed.

**Comment by oddhack** [comment]
@pdaniell-nv I think this is OK to merge per #335, let me know when WG signs off. Also see #346.

**Comment by pdaniell-nv** [comment]
I'll put this on the agenda for the next OpenGL/ES WG meeting January 22nd. If that is too far out let me know and I'll call an email vote.

**Comment by Perksey** [comment]
January 22nd is good for me, I look forward to getting this signed off.

**Comment by pdaniell-nv** [comment]
Approved to merge. Thanks for your patience.

**Comment by oddhack** [comment]
@Perksey a stray BOM crept into gl.xml in this PR which the vk-gl-cts project just got affected by. Propose to pull it out in #506. I assume this was an accident as it's not mentioned anywhere in the comments above but in any event, the file is defined to be encoded in UTF-8. Probably the Python script should be tweaked to open the registry in UTF-8 mode, although it just contains ASCII.

**Pull request merged**